strawberry/types/arguments.py

```diff
@@ -170,6 +170,11 @@ def convert_argument(
         ]
 
     if is_scalar(type_, scalar_registry):
+        from strawberry.relay.types import _GlobalID
+
+        if type_ is _GlobalID:
+            return _GlobalID.from_id(value)
```

Comment:
this would check if the type is a mapped type, but maybe this is fine for now.

I think we should remove the legacy global id support as soon as we can 😊